### PR TITLE
Fixes recursion bug related to enum flags

### LINKED ISSUE #445: [BUG] enum.IntFlag leads to RecursionError
### Description

```py
  File "/test_venv/lib64/python3.11/site-packages/dataclasses_json/core.py", line 393, in <genexpr>
    return list(_asdict(v, encode_json=encode_json) for v in obj)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/test_venv/lib64/python3.11/site-packages/dataclasses_json/core.py", line 393, in _asdict
    return list(_asdict(v, encode_json=encode_json) for v in obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/test_venv/lib64/python3.11/site-packages/dataclasses_json/core.py", line 393, in <genexpr>
    return list(_asdict(v, encode_json=encode_json) for v in obj)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/test_venv/lib64/python3.11/site-packages/dataclasses_json/core.py", line 387, in _asdict
    elif isinstance(obj, Mapping):
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/typing.py", line 1292, in __instancecheck__
    return self.__subclasscheck__(type(obj))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/typing.py", line 1570, in __subclasscheck__
    return issubclass(cls, self.__origin__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen abc>", line 123, in __subclasscheck__
RecursionError: maximum recursion depth exceeded in comparison
```

### Code snippet that reproduces the issue

```python
import enum
from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin


class MyFlag(enum.IntFlag):
#class MyFlag(enum.IntEnum):  # Works
    A = enum.auto()
    C = enum.auto()

@dataclass
class State(DataClassJsonMixin):
    #flag: MyFlag  # fails too
    flag: int

s = State(MyFlag.A)
print(s.to_dict())
```

### Describe the results you expected

Should have worked

### Python version you are using

Python 3.11.3

### Environment description

dataclasses-json==0.5.13
marshmallow==3.20.1
mypy-extensions==1.0.0
packaging==23.1
typing-inspect==0.9.0
typing_extensions==4.7.1

### PR DESCRIPTION
**Pull request by matt035343**
Fixes #445

### CODE REVIEW

File: tests/test_enum.py [hunk 1]
```diff
@@ -71,6 +77,19 @@ class EnumContainer:
     dict_enum_value={"key1str": MyEnum.STR1, "key1float": MyEnum.FLOAT1})
 
 
+@dataclass_json
+@dataclass(frozen=True)
+class FlagContainer:
+    flag_list: List[MyFlag]
+    flag: MyFlag
+
+flag_container_json = '{"flag_list": [1, 4], "flag": 1}'
```
[the remark this answers is not in the document]
Comment (matt035343) [reply tests/test_enum.py:86]: They are binary flags. These fields are integer representations of the bitstrings 001, 010 and 100 respectively.

[the remark this answers is not in the document]
Comment (george-zubrienko) [reply tests/test_enum.py:86]: python never stops amazing me with consistency in naming... `IntFlag`... represents bitstrings. Alright :)

**Review by george-zubrienko — APPROVED** [review]

**Comment by george-zubrienko** [comment]
@powellnorma please take a look as well :)

**Comment by powellnorma** [comment]
Looks good to me

**Pull request merged**